hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/TarContainerPacker.java

```diff
@@ -209,6 +216,12 @@ public void pack(Container<KeyValueContainerData> container,
   private void includePath(Path dir, String subdir,
       ArchiveOutputStream archiveOutput) throws IOException {
 
+    // Add a directory entry before adding files, in case the directory is
+    // empty.
+    ArchiveEntry entry = archiveOutput.createArchiveEntry(dir.toFile(), subdir);
+    archiveOutput.putArchiveEntry(entry);
+
+    // Add files in the directory.
     try (Stream<Path> dirEntries = Files.list(dir)) {
```

Comment:
I think the error is thrown here, so it still needs the CHUNK file to exist?

Reproduced by adding `new File(data.getChunksPath()).delete();` after checking the state of original container.

Reply:
@symious the error is thrown because the chunk directory does not exist. Not having chunk files in the directory should not be a problem here.